osi_driver_inputs.proto

```diff
@@ -0,0 +1,126 @@
+syntax = "proto2";
+
+option optimize_for = SPEED;
+
+import "osi_common.proto";
+import "osi_adas_function.proto";
+
+package osi3;
+
+//
+// \brief An Interface to describe the inputs from a human driver.
+// Contains a base-set of signals with focus on ADAS-functions.
+// The inputs are divided into states and requests.
+//
+message DriverInputs
+{
+    //
+    // The first set of signals are states the driver can (usually) directly set.
+    //
+    optional DriverInitializedStates driver_initialized_states = 1;
+
+    // 
+    // The second set of signals are requests addressed to an external function.
+    //
+    optional DriverRequests driver_requests = 2;
+
+    //
+    // \brief The first set of signals are states the driver can (usually) directly set.
+    //
+    message DriverInitializedStates
+    {
+        // State of the driver seat-belt. It is often an initial condition to start an ADAS-Function.
+        // 0=Open; 1=Closed
+        //
+        optional bool seat_belt = 1;
+
+        // State of the doors. It is often an initial condition to start an ADAS-Function.
+        // 0=Open; 1=Closed
+        //
+        optional bool doors = 2;
+
+        // Hands-On-Detection.
+        // 0=HandsOff; 1=HandsOn
+        //
+        optional bool hands_on_detection = 3;
```

Comment:
ppannusch on 11 Dec 2018  Member
OSI already has SteeringControl for that, which is even more specific (which it should be)

  @ThomasNaderBMW
ThomasNaderBMW on 11 Dec 2018  Author Member
Ou yes, it is part of osi_occupant. Thanks for the hint, this is sth we have to pay attention to.

  @ThomasNaderBMW
ThomasNaderBMW on 13 Dec 2018  Author Member
@CarloVanDriestenBMW & @pmai :
In my opinion the definition of SteeringControl should move from osi_occupant to osi_common, so it can be used in osi_driver_inputs also.
// Definition of hands related to the steering wheel (mostly driver).
//
enum SteeringControl
{
// Hands state is unknown (must not be used in ground truth).
//
STEERING_CONTROL_UNKNOWN = 0;

        // Other (unspecified but known) hand positioning related to the
        // steering wheel.
        //
        STEERING_CONTROL_OTHER = 1;

        // Hands are not on the steering wheel.
        //
        STEERING_CONTROL_NO_HAND = 2;

        // One hand is on the steering wheel. Whether it is the left or right
        // hand is unspecified or unknown.
        //
        // \note If there is no differentiation between one or both hands on the
        // steering wheel, this value should be used.
        //
        STEERING_CONTROL_ONE_HAND = 3;

        // Both hands are on the steering wheel.
        //
        STEERING_CONTROL_BOTH_HANDS = 4;

        // Only left hand is on the steering wheel.
        //
        STEERING_CONTROL_LEFT_HAND = 5;

        // Only right hand is on the steering wheel.
        //
        STEERING_CONTROL_RIGHT_HAND = 6;
    }

Reply:
==> Open Point